landoapi/uplift.py

```diff
@@ -252,45 +299,44 @@ def create_uplift_bug_update_payload(
     return payload
 
 
-UPDATE_RETRIES = 3
-
-
 def update_bugs_for_uplift(
-    bmo_url: str,
-    bmo_api_key: str,
-    changeset_titles: list[str],
     repo_name: str,
-    milestone: int,
+    hgrepo: HgRepo,
+    changeset_titles: list[str],
 ):
     """Update Bugzilla bugs for uplift."""
-    headers = {"X-Bugzilla-API-Key": bmo_api_key}
-    bugs = [str(bug) for title in changeset_titles for bug in parse_bugs(title)]
+    # Parse bug numbers from commits in the stack.
+    bug_ids = [str(bug) for title in changeset_titles for bug in parse_bugs(title)]
+
+    if not bug_ids:
+        raise ValueError("No bugs found in uplift landing.")
+
     params = {
-        "ids": ",".join(bugs),
+        "id": ",".join(bug_ids),
     }
-    bug_endpoint = f"{bmo_url}/rest/bug"
-    resp_get = requests.get(bug_endpoint, headers=headers, params=params)
-    resp_get.raise_for_status()
 
-    bugs = resp_get.json()["bugs"]
+    # Get information about the parsed bugs.
+    bugs = bmo.get_bug(params).json()["bugs"]
+
+    # Get the major release number from `config/milestone.txt`.
+    milestone = get_milestone_major_version(hgrepo)
```

Comment:
As mentioned elsewhere, this method should take a file object or a string/bytes of the file.